lib/elixir/pages/anti-patterns/code-anti-patterns.md

```diff
@@ -47,7 +47,38 @@ Elixir makes a clear distinction between **documentation** and code comments. Th
 
 ## Long parameter list
 
-TODO.
+#### Problem
+
+In a functional language like Elixir, functions tend to be pure and therefore do not access or manipulate values outside their scopes. Since the input values of these functions are just their parameters, it is natural to expect that functions with a long list of parameters be created to keep them pure. However, when this list has **more than three or four parameters**, the function's interface becomes confusing and prone to errors during use.
+
+#### Example
+
+In the following example, the `loan/6` function has an unnecessarily long list of parameters, causing its interface to be confusing and potentially leading developers to introduce errors during calls to this function.
+
+```elixir
+defmodule Library do
+  # Too many parameters that can be grouped!
+  def loan(user_name, email, password, user_alias, book_title, book_ed) do
+    ...
+  end
+end
+```
+
+#### Refactoring
+
+To eliminate this anti-pattern, related parameters can be grouped using `Maps`, `Structs`, or even `Tuples`. This effectively reduces the number of function parameters, simplifying its interface. In the case of the `loan/6`, its parameters were grouped into two different `Maps`, thereby reducing its arity to `loan/2`:
+
+```elixir
+defmodule Library do
+  def loan(%{name: name, email: email, password: password, alias: alias} = user, %{title: title, ed: ed} = book) do
+    ...
+  end
+end
+```
+
+#### Additional remarks
+
+Although this refactoring has grouped parameters using `Maps`, we can find, in different functions, identical sets of parameters that could be grouped. These duplicated sets are known in the literature as *Data Clumps*. In that case, is better to create a `Struct` to group these parameters and reuse this to refactor all functions where these sets of parameters occur.
```

Comment:
```suggestion
Although this refactoring has grouped parameters using maps, we can find, in different functions, identical sets of parameters that could be grouped. These duplicated sets are known in the literature as *Data Clumps*. In that case, is better to create a Struct to group these parameters and reuse this to refactor all functions where these sets of parameters occur.
```

Reply:
Was about to suggest adding a link to [Data Clumps](https://en.wikipedia.org/wiki/Data_clump), and read:

> A data clump, like other [code smells](https://en.wikipedia.org/wiki/Code_smell), can indicate deeper problems with the program design or implementation.

I found it ironic that we're suggesting something which is considered a smell in a guide about avoiding code smells 😅 The context is OOP though so it is quite different, but maybe we don't want to link the suggestion to something considered bad practice?

Reply:
I'm not sure I understood your comment about us suggesting a smell here.  

Long Parameter List and Data Clumps are different traditional code smells, but they can coexist.  In OOP, one way to remove them is to group the parameters of a method in a new class, passing an object of that class in the method call.  This solution reduces the size of the parameter list and still allows reuse, thus eliminating Data Clumps.  What we are suggesting here is basically the same thing, just adapted for Elixir and its features.  

In summary, when we only have a Long Parameter List in a function, we can solve this using only maps.  If we have Data Clumps in several functions, it is better to solve with struct to reuse the grouping that is repeated in these functions.

I don't know if this raised any questions, but just to be clear, a struct is not a Data Clumps.  If the metadata of a struct is used together as individual parameters of several functions, then we will have a Data Clumps.  Example: name, age and gender used together as individual parameters of several different functions is a Data Clumps.  It would be better to define a %User{} containing this group and reuse this definition in parameter lists where this set is repeated.

Reply:
Oh sorry I misread it as "theses maps/structs/tuples are referred to as data clumps", which is the opposite of what you were saying.
Sorry for the confusion 🙇

Reply:
is this guide just for the grouping? the below issue is something i encounter more frequently

>  Although such cases are rare, it is possible for a method to legitimately take half a dozen or more completely unrelated parameters that could not be cleanly turned into a single object. This, however, suggests that the method is trying to do far too much and would be better broken into multiple methods, each of which is responsible for a smaller piece of the overall responsibility.

Reply:
> is this guide just for the grouping? the below issue is something i encounter more frequently
> 
> >  Although such cases are rare, it is possible for a method to legitimately take half a dozen or more completely unrelated parameters that could not be cleanly turned into a single object. This, however, suggests that the method is trying to do far too much and would be better broken into multiple methods, each of which is responsible for a smaller piece of the overall responsibility.
> 
> 

You are referring to the lack of cohesion, which can also be one of the causes for a function that has "Long Parameter List".  As there are several possible causes for this smell, I don't know if it's worth trying to delve too deeply into them.  I think it is more interesting to present only the problem and a possible solution, without the ambition of trying to cover all possible contexts (or causes).  This is usually how smell catalogs are structured.